### PR TITLE
Pool withdraw op

### LINKED ISSUE #3105: Implement LiquidityPoolWithdrawOpFrame for CAP-38
The following steps are required when implementing `LiquidityPoolWithdrawOpFrame`:
- Implement a utility function to compute withdrawal amounts (also needed to handle authorization revocation elsewhere) (#3147)
- Update `ConservationOfLumens` invariant (#3153)
- Implement `LiquidityPoolWithdrawOpFrame` itself (#3147)

### PR DESCRIPTION
**Pull request by sisuresh**
# Description

Resolves #3105.

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

**Review by jonjove — COMMENTED** [review]
Looks really good, just a couple of nitpicks.

**Review by MonsieurNicolas — APPROVED** [review]
LGTM!

File: src/transactions/test/LiquidityPoolWithdrawTests.cpp [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+#include "test/TestExceptions.h"
+#include "test/TestUtils.h"
+#include "test/TxTests.h"
+#include "test/test.h"
+#include "transactions/TransactionUtils.h"
+
+using namespace stellar;
+using namespace stellar::txtest;
+
+TEST_CASE("liquidityPoolWithdraw", "[tx][liquiditypool]")
```
[the remark this answers is not in the document]
Comment (sisuresh) [reply src/transactions/test/LiquidityPoolWithdrawTests.cpp]: Done

File: src/transactions/test/LiquidityPoolWithdrawTests.cpp [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        for_versions_to(17, *app, [&] {
+            REQUIRE_THROWS_AS(acc1.liquidityPoolWithdraw(PoolID{}, 0, 0, 0),
+                              ex_opNOT_SUPPORTED);
+        });
+    }
+
+    for_versions_from(18, *app, [&] {
+        SECTION("malformed")
+        {
+
```
[the remark this answers is not in the document]
Comment (sisuresh) [reply src/transactions/test/LiquidityPoolWithdrawTests.cpp]: Fixed

File: src/test/TxTests.h [hunk 3]
```diff
@@ -80,6 +80,9 @@ void validateTxResults(TransactionFramePtr const& tx, Application& app,
                        ValidationResult validationResult,
                        TransactionResult const& applyResult = {});
 
+void checkLiquidityPool(Application& app, PoolID const& poolID,
+                        int64_t reserveA, int64_t reserveB,
+                        int64_t totalPoolShares);
```
[the remark this answers is not in the document]
Comment (sisuresh) [reply src/test/TxTests.h:85]: Fixed

**Comment by jonjove** [comment]
r+ 536465a

**Pull request merged**